vllm/v1/core/single_type_kv_cache_manager.py

```diff
@@ -532,29 +532,149 @@ def find_longest_cache_hit(
-        return new_blocks
+                            num_tokens: int) -> list[KVCacheBlock]:           
+        if self.kv_cache_spec.cache_strategy == "disabled":
+            new_blocks = super().allocate_new_blocks(request_id, num_tokens)
+            assert len(self.req_to_blocks[request_id]) == 1, (
+                "MambaManager should only allocate 1 block for each request.")
+            return new_blocks
+
+        req_blocks = self.req_to_blocks[request_id]
+        num_required_blocks = cdiv(num_tokens, self.block_size)
+        num_new_blocks = num_required_blocks - len(req_blocks)
+        if num_new_blocks <= 0:
+            return []
+        else:
+            if num_new_blocks > 2 and self.kv_cache_spec.cache_strategy == "last":
+                # for the last strategy only - allocate 2 blocks: 
+                #  one for block_size aligned state
+                #  and one for the last temporary state
+                new_blocks = self.block_pool.get_new_blocks(2)
+            else:
+                new_blocks = self.block_pool.get_new_blocks(num_new_blocks)
+            req_blocks.extend(new_blocks)
+            return new_blocks        
+
+    def get_num_blocks_to_allocate(
+            self, request_id: str, num_tokens: int,
+            new_computed_blocks: list[KVCacheBlock]) -> int:
+        """
+        Get the number of blocks needed to be allocated for the request.
+
+        Args:
+            request_id: The request ID.
+            num_tokens: The total number of tokens that need a slot (including 
+                tokens that are already allocated).
+            new_computed_blocks: The new computed blocks just hitting the
+                prefix caching.
+
+        Returns:
+            The number of blocks.
+        """
+
+        num_required_blocks = cdiv(num_tokens, self.block_size)
+        num_new_blocks = (num_required_blocks - len(new_computed_blocks) -
+                          len(self.req_to_blocks[request_id]))
+        # If a computed block of a request is an eviction candidate (in the
+        # free queue and ref_cnt == 0), it will be changed from a free block
+        # to a computed block when the request is allocated, so we also count
+        # it as needed to be allocated.
+        num_evictable_computed_blocks = sum(
+            blk.ref_cnt == 0 and not blk.is_null
+            for blk in new_computed_blocks)
+        return num_new_blocks + num_evictable_computed_blocks
+
+    def cache_blocks(self, request: Request, block_hashes: list[BlockHash],
+                     num_tokens: int) -> None:
+        """
+        Cache the blocks for the request.
+
+        Args:
+            request: The request.
+            block_hashes: The block hashes of the request.
+            num_tokens: The total number of tokens that need to be cached 
+                (including tokens that are already cached).
+        """
 
+        #TODO: Just copied parent class implementation here to verify logic.
```

Comment:
Could you please revert this (e.g., don't copy any implementation from parent class if there is no reason to modify the behaviour) to facilitate code review?